api/osf_api.py

```diff
@@ -784,3 +784,51 @@ def get_user_developer_app_data(session, app_id=None):
     if data:
         return data
     return None
+
+
+def create_personal_access_token(
+    session, name='OSF Test PAT', scopes='osf.nodes.full_read'
+):
+    """Create a Personal Access Token for the user that is currently logged in to OSF
+    (via session object). Default scope is 'osf.nodes.full_read'. The scopes parameter
+    is a string (not a list) with multiple scopes separated by a space.
+    EX: 'osf.users.profile_write osf.full_write osf.nodes.full_write'
+    """
+    if not session:
+        session = get_default_session()
+    url = '/v2/tokens/'
+    raw_payload = {
+        'data': {
+            'type': 'tokens',
+            'attributes': {
+                'name': name,
+                'scopes': scopes,
+            },
+        }
+    }
+    return_data = session.post(
+        url=url, item_type='tokens', raw_body=json.dumps(raw_payload)
+    )
+    # Return the public id and the private token_id
+    if return_data:
+        public_id = return_data['data']['id']
+        token_id = return_data['data']['attributes']['token_id']
+        return [public_id, token_id]
+    return None
+
+
+def delete_personal_access_token(session, token_id=None):
+    """Delete a User's Personal Access Token as identified by its token id."""
+    if not session:
+        session = get_default_session()
+    url = '/v2/tokens/{}/'.format(token_id)
```

Comment:
I am very very surprised that the `token_id` is used in URL ... very vulnerable ... and not just the delete request. This is a security bug that we (Devs) need to fix (@mfraezz for transparency) unless this maybe a deprecated way?